src/ckb/channel.rs

```diff
@@ -2831,33 +2907,53 @@ impl ChannelActorState {
         let commitment_lock_script =
             commitment_lock_ctx.get_commitment_lock_script(&blake2b_256(witnesses)[0..20]);
 
-        let outputs = vec![
-            CellOutput::new_builder()
-                .capacity((immediately_spendable_value as u64).pack())
-                .lock(immediate_secp256k1_lock_script)
-                .build(),
-            CellOutput::new_builder()
-                .capacity((time_locked_value as u64).pack())
-                .lock(commitment_lock_script)
-                .build(),
-        ];
-        let outputs_data = vec![Bytes::default(); outputs.len()];
+        if let Some(udt_type_script) = &self.funding_type_script {
+            let immediate_output = packed::CellOutput::new_builder()
+                .capacity(Capacity::shannons(6100000000).pack())
```

Comment:
I think the commitment transaction built from here is unspendable because it didn't add the required sudt cell deps. Can you please rebase this PR to the main branch. The main branch has a unit test `test_create_channel` which uses `ckb-testtool` to verify the validity of the commitment transaction https://github.com/contrun/ckb-pcn-node/blob/1b22666d295f8a8a4b07c0035a681bf1c6907ce7/src/ckb/channel.rs#L3435-L3635 .